routes/signUp/helper.js

```diff
@@ -63,16 +63,17 @@ module.exports = {
                 userId: user._id,
             });
 
-            if(isProducer){
+            /* if(isProducer){
                 await producer.save();
-            } 
+            }  */
+            await producer.save();
```

Comment:
this is still wrong, you cannot create a producer on every `signUp`. use the variable `typesofProducts` to gate whether a produce should be created of not